### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by lzan98**
- Added extension for empty list in delete use case
- Modified 'help' use case to high priority
- Removed 'hide private contact details' from Glossary

#45

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -329,24 +329,27 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 3a. The list is empty.
+    
+    Use case ends
+* 3b. The given student number is invalid or does not exist in the system
 
-    * 3a1. TAlent Assistant™ shows an error message.
+    * 3a1. TAlent Assistant™ displays an error message
 
-  Use case resumes at step 2.
+  Use case resumes at Step 2.
```
Comment (LeeEnEn) [inline docs/DeveloperGuide.md:352]: Perhaps `Step` should remain as lower case? 🤔

Comment (lzan98) [reply docs/DeveloperGuide.md]: Thanks for spotting the mistake! Fixed.

**Review by LeeEnEn — APPROVED** [review]
LGTM! 🥂

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (LeeEnEn) [reply docs/DeveloperGuide.md]: 👍

**Pull request merged**